test/unit/resources/servers.js

```diff
@@ -0,0 +1,674 @@
+      return this.backdoor({
+        command: 'moveSystemFoldersUnderneathInbox'
+      });
+    },
+
+    /**
+     * ActiveSync-Only!
+     *
+     * Ask the ActiveSync server for the list of distinct device id's it has seen
+     * since startup or when the clear option was last provided.
+     *
+     * @param {Boolean} [opts.clear]
+     *   Clear the list subsequent to returning the current list contents.
+     */
+    getObservedDeviceIds: function(opts) {
+      return this.backdoor({
+        command: 'getObservedDeviceIds',
+        clearObservedDeviceIds: opts && opts.clear
+      });
+    },
+
+    //////////////////////////////////////////////////////////////////////////////
+    // OAuth stuff
+    //
+    // This should all probably be on a separate helper object.
+    // Another thing for the great test refactoring/cleanup.
+    _oauthbackdoor: function(request, explicitPath) {
+      var xhr = new XMLHttpRequest({mozSystem: true, mozAnon: true});
+      xhr.open('POST', this.serverInfo.oauthInfo.backdoor, false);
+      xhr.send(JSON.stringify(request));
+      var response = xhr.response || null;
+      try {
+        if (response)
+          response = JSON.parse(response);
+      }
+      catch (ex) {
+        logic(this, 'backdoorError', {
+          request: request,
+          response: response
+        });
+        return null;
+      }
+      logic(this, 'backdoor', {
+        request: request,
+        response: response
+      });
+      return response;
+    },
+
+    oauth_getNumAccessTokensProvided: function(params) {
+      return this._oauthbackdoor({
+        command: 'getNumAccessTokensProvided',
+        reset: params.reset
+      });
+    }
+
+  }
+
+  function extractUsernameFromEmail(str) {
+    var idx = str.indexOf('@');
+    if (idx === -1)
+      return str;
+    return str.substring(0, idx);
+  }
+
+
+
+  var MONTHS = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep',
+                'Oct', 'Nov', 'Dec'];
+
+  /**
+   * Format an IMAP date-time or an 2822 date-time string.  The difference is
+   * whether the date has dashes between it or spaces.
+   *
+   * An RFC 3501 example is: 17-Jul-1996 02:44:25 -0700
+   * An RFC 2822 example is: 21 Nov 1997 10:05:43 -0600
+   */
+  function formatDateTime(date, format, forceTZOffset) {
+    var s;
+    var dateSpacer = (format === 'imap') ? '-' : ' ';
+    var tzOffset = (forceTZOffset != null) ? forceTZOffset :
+          date.getTimezoneOffset();
+    s = ((date.getDate() < 10) ? ' ' : '') + date.getDate() + dateSpacer +
+      MONTHS[date.getMonth()] + dateSpacer +
+      date.getFullYear() + ' ' +
+      ('0'+date.getHours()).slice(-2) + ':' +
+      ('0'+date.getMinutes()).slice(-2) + ':' +
+      ('0'+date.getSeconds()).slice(-2) +
+      ((tzOffset > 0) ? ' -' : ' +' ) +
+      ('0'+(Math.abs(tzOffset) / 60)).slice(-2) +
+      ('0'+(Math.abs(tzOffset) % 60)).slice(-2);
+    return s;
+  }
+
+
+
+
+
+
+
```

Comment:
Are these clusters of newlines (there was at least one other one) intentional for styling, or just refactoring fallout?  (Or just Riker napping on the keyboard? :)